File: src/main/scala/com/kapilkoju/autopos/catalog/api/BrandApi.scala
package com.kapilkoju.autopos.catalog.api

import com.kapilkoju.autopos.catalog.repo.BrandRepo
import com.kapilkoju.autopos.catalog.repo.BrandRepo
import com.kapilkoju.autopos.kernel.api.BaseApi
import org.springframework.web.bind.annotation.{RequestMapping, RestController}


@RestController
@RequestMapping(Array("/api/brands"))
class BrandApi(private val brandRepository: BrandRepo)
  extends BaseApi(brandRepository, "brand", "/api/brands")

File: src/main/kotlin/com/kapilkoju/autopos/catalog/api/BrandApi.kt
package com.kapilkoju.autopos.catalog.api

import com.kapilkoju.autopos.catalog.domain.Brand
import com.kapilkoju.autopos.catalog.service.BrandRepo
import com.kapilkoju.autopos.kernel.api.BaseApi
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping(BrandApi.baseUrl)
class BrandApi(brandRepo: BrandRepo) : BaseApi<Brand>(brandRepo, "brand", BrandApi.baseUrl) {

    companion object {
        const val baseUrl = "/api/brands"
    }
}